ecosystem/sep-0024.md

```diff
@@ -305,7 +305,7 @@ The basic parameters are summarized in the table below.
 
 Name | Type | Description
 -----|------|------------
-`callback` | string | (optional) A URL that the anchor should `POST` a JSON message to when the user successfully completes the interactive flow. Can also be set to `postMessage`.
+`callback` | string | (optional) A URL that the anchor should `POST` a JSON message to when the `status` property changes, or when the `kyc_verified` property changes and the `status` property is incomplete or pending. Can also be set to `postMessage`.
```

Comment:
```suggestion
`callback` | string | (optional) A URL that the anchor should `POST` a JSON message to when the `status` or `kyc_verified` properties change. Can also be set to `postMessage`.
```

Reply:
Prior to this change if the client uses `postMessage` as the callback they expect to get a single call to postMessage when the interactive flow has changed. With this change an implementer might think they should call postMessage during the interactive flow resulting in multiple `postMessage` calls. It's likely because of the previous definition that a client has implemented their application to close the window containing the interactive flow when the first postMessage is received.

I think because of this we need to retain some of the original language so that when postMessage is used the first call to postMessage indicates not only a change in status or properties but that the interactive flow has completed.

Reply:
+1 to Leigh's comment, I didn't look at the original language before making my comment.

Reply:
@leighmcculloch I'm not quite understanding what you're suggesting? It seems like our proposed change will be a breaking one for your specific case, and I don't think there's any way around it. Unless you're saying that postMessage would behave differently than callback URLs?

Reply:
@morleyzhi I think what Leigh is saying is that the original purpose to the `postMessage` call was for the client to close the window. There's no reason to remove this language, since the change we're making is additive. The first `postMessage` callback can signal that the flow is complete AND subsequent callbacks can signal a change in `status` or `kyc_verified`.

Reply:
Note that when the flow completes, the `status` will likely change too, so the distinction between the two purposes is actually not needed from a functional standpoint.

Reply:
The problem is that there may be some status changes before the flow is complete: the incomplete -> pending_kyc one comes to mind.

Reply:
If we're concerned about backwards compatibility, we can add a second callback property.

Reply:
That may be necessary, because if you look at the SEP, `callback` is stated to be used to signal the end of the interactive flow all over the place. Going back on this would cause issues for wallets expecting a single callback request.

Reply:
`callback_on_change` sounds good to me.